packages/git/src/common/git-model.ts

```diff
@@ -65,6 +70,7 @@ export namespace WorkingDirectoryStatus {
                 && (left.aheadBehind ? left.aheadBehind.ahead : -1) === (right.aheadBehind ? right.aheadBehind.ahead : -1)
                 && (left.aheadBehind ? left.aheadBehind.behind : -1) === (right.aheadBehind ? right.aheadBehind.behind : -1)
                 && left.changes.length === right.changes.length
+                && (left.incomplete === !!right.incomplete || !!left.incomplete === right.incomplete)
```

Comment:
I am either blind or it is still not complete. It should be `true` is both left and right are `undefined`.

Please take the time and write a test for it, it would be something like this:
```
    it('yes', () => {
        ([
            [undefined, undefined],
            [undefined, true],
            [undefined, false],
            [true, undefined],
            [true, true],
            [true, false],
            [false, undefined],
            [false, true],
            [false, false]
        ] as [boolean | undefined, boolean | undefined][]).forEach(([left, right]) => {
            if (left === !!right || !!left === right) {
                console.log('left', left, 'right', right);
            }
        });
    });
```